swapi/store.js

```diff
@@ -43,6 +55,32 @@ const reducer = (state = {}, action) => {
 				operations: state.operations - 1,
 			}
 		}
+
+		case types.EXPAND_ITEM: {
+			const itemDisplay = action.itemDisplay;
+			const expandAll = action.expandAll;
+
+			var data = state.data[state.endpoint];
+			data = data.map((item) => {
```

Comment:
På den her måde bliver hvert item i listen ændret, som gør at React vil rerender alle items. Du kunne vælge kun at lave et nyt object for det item som man expander, og bare returnere dem som dermed ikke bliver ændret.

Men en lidt smartere løsning kunne være ikke at gemme `isExpanded` på selve item, men i en object eller liste for sig. Din store kunne således ser sådan ud:

```
{
   data: {
       // data som den er nu
   },
   expandedItemStatus: {
      [itemUrl]: true,
      [otherItemUrl]: false,
   },
```